### PR TITLE
ci: Add scripts to run cc tests

### PR DESCRIPTION
**Pull request by GabyCT**
This will install the cc-test repository as well as all the
dependencies for the functional tests.

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: .ci/install_cc_tests.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#
+
+#!/bin/bash
+
+set -e
+
+echo "Retrieve cc-tests repository"
+go get -d github.com/clearcontainers/tests
+
+echo "Retrieve cc-tests dependencies"
```
Comment (dlespiau) [inline .ci/install_cc_tests.sh]: This shouldn't be needed, we should vendor all tests dependencies in tests/.

File: .ci/run_cc_tests.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2017 Intel Corporation
```
Comment (dlespiau) [inline .ci/run_cc_tests.sh]: This is script that will be common to all testing, should be in the `tests` repo.

File: .ci/install_cc_tests.sh (hunk 1 above)
Comment (devimc) [inline .ci/install_cc_tests.sh]: but ```ginko``` binary is needed to run our tests

File: .ci/install_cc_tests.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#!/bin/bash
+
+set -e
+
+echo "Retrieve cc-tests repository"
+go get -d github.com/clearcontainers/tests
```
Comment (jodh-intel) [inline .ci/install_cc_tests.sh]: You can remove the `go get`'s below by making this:

```
go get -d github.com/clearcontainers/tests/...
```

File: .ci/run_cc_tests.sh (hunk 2 above)
Comment (devimc) [inline .ci/run_cc_tests.sh]: or we can just run ```make functional``` in semaphore

Comment (dlespiau) [inline .ci/run_cc_tests.sh]: I'm guessing there will be more things at some point, make docker, deal with coverage data, .... May as well have one single script as our "API" and never change that.

File: .ci/install_cc_tests.sh (hunk 1 above)
Comment (dlespiau) [inline .ci/install_cc_tests.sh]: ah, hadn't realised that.

File: .ci/run_cc_tests.sh (hunk 2 above)
Comment (GabyCT) [inline .ci/run_cc_tests.sh]: @dlespiau ok thanks let me modify it

File: .ci/install_cc_tests.sh (hunk 1 above)
Comment (dlespiau) [inline .ci/install_cc_tests.sh]: In any case, this should be part of the `tests` repo. proxy testing will need it as well for instance.

**Comment by GabyCT** [comment]
@dlespiau, changes are updated in this PR and I will do the changes to the ```tests``` repo

File: .ci/install_cc_tests.sh (hunk 3 above)
Comment (GabyCT) [inline .ci/install_cc_tests.sh]: @jodh-intel, I tried it in that way however ginkgo is not download it

**Comment by dlespiau** [comment]
There are a few questions left:

- It doesn't seem very clear that we should install the `tests` repo in `install_runtime.sh`. `install_runtime.sh` should just install the runtime, nothing else, I think.
- We don't call into anything in the `tests` repo, so don't run anything atm. Can we all agree to have a script in the `tests` repo all package have to call to run the tests. 

So what about having a script called "run_tests.sh" that will:

1. git clone the `tests` repo
2. call a script in the `tests` repo to run the tests

**Comment by dlespiau** [comment]
Something that could help thinking of a good design for this is thinking about running the tests as an API call or an interface. The `tests` repo provides an "API" to the other packages. One of the calls is "run the tests". We don't need to know about the details of running the tests in this repository, ie, we shouldn't need to install ginkgo here, that's an implementation detail of the `tests` repository.

So now the problem becomes defining a simple API that can be called from runtime, proxy, and shim to setup the test environment and run the tests while sharing the maximum amount of code possible.

**Comment by chavafg** [comment]
Agree with @dlespiau comment above. 
We should call `install_cc_tests.sh` on the project settings on SemaphoreCI. 

But we should also remove these entries:
```
bash .ci/install_cc_proxy.sh

bash .ci/install_cc_shim.sh

bash .ci/install_virtcontainers.sh
```
from `install_runtime.sh` and add them to SemaphoreCI settings.

Thoughts?

**Comment by jodh-intel** [comment]
I'm personally wary of having complex settings for Semaphore since as it doesn't support having a config file like Travis, it is rather too easy to inadvertently remove / re-order scripts using their web interface (I've been bitten by that before).

Although we aren't actually using Semaphore for https://github.com/01org/clr-oci-runtime, I did rework the CI scripts to support it and the intention was always to call *at most* 3 scripts (for any CI) as shown in the Travis config:

https://github.com/01org/cc-oci-runtime/blob/master/.travis.yml#L18..L25

**Comment by devimc** [comment]
I agree with @jodh-intel 
lets keep it simple (KISS) creating 3 scripts 

* setup.sh to install packages and setup the environment
* build.sh or run.sh to test the PR
* cleanup.sh or teardown.sh to report or clean something

currently ```install_runtime.sh``` installs proxy, shim and virtcontainers and it does not make sense

**Comment by GabyCT** [comment]
@dlespiau, @jodh-intel, @devimc and @chavafg, I created this issue https://github.com/clearcontainers/runtime/issues/196 where we will have only 3 scripts in our CI but as this will need a re-work, can we just merged this PR in order to run the functional tests meanwhile?

**Comment by devimc** [comment]
lgtm

**Comment by dlespiau** [comment]
I must be missing something, we won't be running the functional tests with this PR?

**Comment by GabyCT** [comment]
@dlespiau at the clearcontainers/tests I will add in the .ci a script that will run the functional tests so once this PR is merged in semaphore I will run this script (as part of the Jobs).  I need this PR to be merged in order to retrieve the clearcontainers/tests repository.

**Pull request merged**